customskill/request/request.go

```diff
@@ -0,0 +1,44 @@
+package request
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+var jsonUnmarshal = json.Unmarshal // Used to enable unit testing
```

Comment:
how does this work exactly? I've never seen this done before.

Reply:
By using a variable for the `json.Unmarshal` function (instead of the function itself) we can mock it out for unit testing.